### ISSUE TITLE
Query on select field with integer values doesn't work anymore

### ISSUE THREAD

**Issue by jordanaubert** [bolt/core#3019]
I can't fetch Content using Query service on select field with integer values.

Details
-------

| Question                                              | Answer
|------------------------------------------------------ | ------
| Relevant Bolt Version                                 | 5.0.7
| Install type                                          |  composer install
| <abbr title="Backwards Compatibility">BC</abbr> Break | **yes**
| PHP version                                           | 7.4 |
| Web server                                            | Nginx

Reproduction
------------
I have this content type defined in my config file :

```yaml
contenttype:
    name: contenttype
    fields:
        select_field:
            type: select
            values:
                0: value1
                301: value2
                302: value3
```

I want to fetch some data with following request using Query service :

```php
private Bolt\Storage\Query $$query;

public function __construct(Bolt\Storage\Query $query) {
    $this->query = $query;
}

$this->query->getContent('contenttype', ['select_field' => 301]);
```

No result returned while I can see existing value in database :

<img width="667" alt="Capture d’écran 2021-12-16 à 11 28 59" src="https://user-images.githubusercontent.com/16880485/146354794-c6d6462f-ea46-4de6-a2a6-75b2d59b917e.png">

It's possible to fetch data when values are stored as string instead of integer :
<img width="667" alt="Capture d’écran 2021-12-16 à 11 30 59" src="https://user-images.githubusercontent.com/16880485/146355022-53305aaf-5c20-4057-8130-1b98505497f9.png">

It's a BC break because the query worked on Bolt `4.2.5`

**Comment by jordanaubert** [comment]
This is probably why we can't query json field with integer values :

> [JSON_SEARCH(json_doc, one_or_all, search_str[, escape_char[, path] ...])](https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-search)
> Returns the path to the _**given string**_ within a JSON document.

It seems that by design, `JSON_SEARCH` works with strings only.

see also https://bugs.mysql.com/bug.php?id=79233

**Comment by jordanaubert** [comment]
Issue fixed by #3302
